### PR TITLE
Add 3Des to unit test.

### PR DESCRIPTION
**Pull request by jrblixt**
Adds the 3Des unit test functions and updates parameter checking where needed.

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -3314,6 +3318,294 @@ static int test_wc_Sha384Final (void)
...
+{
+#ifndef NO_DES3
+    Des3 des;
+
+    #ifndef WOLFSSL_SMALL_STACK
+        byte cipher[24];
+        byte plain[24];
+    #elif defined(WOLFSSL_SMALL_STACK)
+        byte* cipher = (byte*) malloc(sizeof(byte) * 24);
+        byte* plain = (byte*) malloc(sizeof(byte) * 24);
```
Comment (cconlon) [inline tests/api.c]: These two calls should probably be using XMALLOC() instead of malloc(), right?

File: tests/api.c [hunk 2]
```diff
@@ -3314,6 +3318,294 @@ static int test_wc_Sha384Final (void)
...
+    if (ret == 0) {
+        ret = wc_Des3_CbcEncrypt(NULL, cipher, vector, 24);
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcEncrypt(&des, NULL, vector, 24);
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcEncrypt(&des, cipher, NULL, sizeof(vector));
+        }
+        if (ret != BAD_FUNC_ARG) {
+            printf("\nReturn code: %d\n", ret);
```
Comment (cconlon) [inline tests/api.c]: Should this print statement be here, or was this left over from debugging?

File: tests/api.c [hunk 3]
```diff
@@ -3314,6 +3318,294 @@ static int test_wc_Sha384Final (void)
...
+    if (ret == 0) {
+        ret = wc_Des3_CbcDecrypt(NULL, plain, cipher, 24);
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcDecrypt(&des, NULL, cipher, 24);
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcDecrypt(&des, plain, NULL, 24);
+        }
+        if (ret != BAD_FUNC_ARG) {
+            printf("\nReturn code: %d\n", ret);
```
Comment (cconlon) [inline tests/api.c]: Same as above, should this print statement be here?

File: tests/api.c [hunk 4]
```diff
@@ -3314,6 +3318,294 @@ static int test_wc_Sha384Final (void)
...
+
+    word32 vectorSz, cipherSz;
+    int ret;
+
+    #ifndef WOLFSSL_SMALL_STACK
+        byte cipher[24];
+        byte plain[24];
+    #elif defined(WOLFSSL_SMALL_STACK)
+        byte* cipher = (byte*) malloc(sizeof(byte) * 24);
+        byte* plain = (byte*) malloc(sizeof(byte) * 24);
```
Comment (cconlon) [inline tests/api.c]: Like above, these should probably be XMALLOC() calls instead of malloc(), for portability.

File: tests/api.c [hunk 5]
```diff
@@ -3314,6 +3318,294 @@ static int test_wc_Sha384Final (void)
...
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcEncryptWithKey(cipher, vector, vectorSz, NULL, iv);
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcEncryptWithKey(cipher, vector, vectorSz,
+                                                                    key, NULL);
+        } else {
+            /* Return code catch. */
+            printf("\nReturn code: %d\n", ret);
```
Comment (cconlon) [inline tests/api.c]: Should this print statement be here?

File: tests/api.c [hunk 6]
```diff
@@ -3314,6 +3318,294 @@ static int test_wc_Sha384Final (void)
...
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcDecryptWithKey(plain, NULL, cipherSz, key, iv);
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcDecryptWithKey(plain, cipher, cipherSz, NULL, iv);
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcDecryptWithKey(plain, cipher, cipherSz, key, NULL);
+        } else {
+            printf("\nReturn code: %d\n", ret);
```
Comment (cconlon) [inline tests/api.c]: Should this print statement be here?

**Comment by kaleb-himes** [comment]
retest this please jenkins

**Comment by jrblixt** [comment]
retest this please Jenkins.

**Comment by cconlon** [comment]
Thanks!

**Pull request merged**